webapp/channels/src/components/file_preview_modal/file_preview_modal.tsx

```diff
@@ -164,21 +164,44 @@ export default class FilePreviewModal extends React.PureComponent<Props, State>
         }
     };
 
+    isImageUrl = (url: string): boolean => {
+        const fileType = Utils.getFileType(url);
+        return fileType === FileTypes.IMAGE || fileType === FileTypes.SVG;
+    };
+
     loadImage = (index: number) => {
         const fileInfo = this.props.fileInfos[index];
         if (isFileInfo(fileInfo) && fileInfo.archived) {
             this.handleImageLoaded(index);
             return;
         }
-        const fileType = Utils.getFileType(fileInfo.extension);
 
-        if (fileType === FileTypes.IMAGE && isFileInfo(fileInfo)) {
-            let previewUrl;
-            if (fileInfo.has_preview_image) {
-                previewUrl = getFilePreviewUrl(fileInfo.id);
-            } else {
-                // some images (eg animated gifs) just show the file itself and not a preview
-                previewUrl = getFileUrl(fileInfo.id);
+        // Determine file type based on the object type
+        let fileType;
+        if (isFileInfo(fileInfo)) {
+            fileType = Utils.getFileType(fileInfo.extension);
+        } else if (isLinkInfo(fileInfo)) {
+            // For LinkInfo, pass the full URL to getFileType to leverage enhanced URL detection
+            fileType = Utils.getFileType(fileInfo.link);
+        } else {
+            fileType = Utils.getFileType('');
```

Comment:
nit: should we just leave `fileType` undefined or set it explicitly to some unknown constant? (Seems weird to ask `getFileType` for `''`.)

Reply:
well, I see your point, I kept it there since the function has an early return it was basically the same. But yes, could be confusing. So, using the `FileTypes.OTHER` directly